src/Kestrel.Core/Internal/Infrastructure/ConnectionManagerShutdownExtensions.cs

```diff
@@ -8,29 +8,31 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure
 {
-    public static class HttpConnectionManagerShutdownExtensions
+    public static class ConnectionManagerShutdownExtensions
     {
-        public static async Task<bool> CloseAllConnectionsAsync(this HttpConnectionManager connectionManager, CancellationToken token)
+        public static async Task<bool> CloseAllConnectionsAsync(this ConnectionManager connectionManager, CancellationToken token)
         {
             var closeTasks = new List<Task>();
 
             connectionManager.Walk(connection =>
             {
-                closeTasks.Add(connection.StopProcessingNextRequestAsync());
+                connection.TransportConnection.Close();
+                closeTasks.Add(connection.ExecutionTask);
             });
 
             var allClosedTask = Task.WhenAll(closeTasks.ToArray());
             return await Task.WhenAny(allClosedTask, CancellationTokenAsTask(token)).ConfigureAwait(false) == allClosedTask;
         }
 
-        public static async Task<bool> AbortAllConnectionsAsync(this HttpConnectionManager connectionManager)
+        public static async Task<bool> AbortAllConnectionsAsync(this ConnectionManager connectionManager)
         {
             var abortTasks = new List<Task>();
             var canceledException = new ConnectionAbortedException(CoreStrings.ConnectionAbortedDuringServerShutdown);
 
             connectionManager.Walk(connection =>
             {
-                abortTasks.Add(connection.AbortAsync(canceledException));
+                connection.TransportConnection.Abort(canceledException);
+                abortTasks.Add(connection.ExecutionTask);
```

Comment:
This used to wait on just the socket being closed and not the middleware completing. On my machine, this is causing GracefulTurnsAbortiveIfRequestsDoNotFinish to consistently fail since NotAllConnectionsAborted now get logged instead of just NotAllConnectionsClosedGracefully. I'm not sure you you managed to get the CI green.

Reply:
It waits on both https://github.com/aspnet/KestrelHttpServer/blob/3122bdfc7c85781410d423929aff24c1f30136c3/src/Kestrel.Core/Internal/ConnectionDispatcher.cs#L82. Not sure what you mean.

Reply:
You're right. It's waiting longer than before causing an unexpected log failing the  GracefulTurnsAbortiveIfRequestsDoNotFinish test. This test isn't running on Jenkins since it only runs on Win 10 and above. The Jenkins Windows agent is running Win 8.